src/rpc/server.cpp

```diff
@@ -137,8 +147,9 @@ UniValue ValueFromAmount(const CAmount& amount)
 {
     bool sign = amount < 0;
     int64_t n_abs = (sign ? -amount : amount);
-    int64_t quotient = n_abs / COIN;
-    int64_t remainder = n_abs % COIN;
+    unsigned int nRpcUnit = pow(10, nRPCAmountDecimals);
```

Comment:
Yes, floating point calculations tend to be unpredictable. You could maybe use
```cpp
static constexpr std::array<CAmount,9> pow_10 {{1,10,100,1000,10000,100000,1000000,10000000,100000000}};
nRpcUnit = pow_10.at(nRPCAmountDecimals);
```
or come up with something else.